src/modules/extra/m_ssl_openssl.cpp

```diff
@@ -463,45 +463,62 @@ class OpenSSLIOHook : public SSLIOHook
 	// Returns 1 if handshake succeeded, 0 if it is still in progress, -1 if it failed
 	int Handshake(StreamSocket* user)
 	{
+		LocalUser* luser = ((UserIOHandler*)user)->user;
```

Comment:
What about SSL connections which aren't user connections (e.g. m_httpd) that won't use UserIOHandler?

Reply:
Test

Reply:
I would have assumed that a "user" would be anything that uses the connection, regardless of whether or not it is a person.  Is 'LocalUser' then not appropriate either?  What else should I use?  I need something that I can get the IP, port, Subject and Issuer Distinguished Name, and fingerprint from.  Alternatively, is there a guide that explains the abstractions within InspIRCd?  Also, could you recommend a FOSS CLI tool for perusing C++ code?  cscope only works with C :(

Reply:
`LocalUser` represents an IRC client which is directly connected to the local server.

`SSLIOHook::IsSSL` might do what you want?